packages/contracts-ops/contracts/test/RebootVerify.t.sol

```diff
@@ -0,0 +1,256 @@
+            assertEq(cnMgr.domainToReplica(connectionDomain), replica);
+        }
+
+        // Verify new implementations have been set
+
+        // Verify home implementation
+        address storedHomeImpl = vm
+            .load(address(homeUpgrade(domain).beacon), bytes32(0))
+            .bytes32ToAddress();
+        assertEq(storedHomeImpl, address(homeImpl));
+
+        // Verify Replica implementation
+        string memory remote = getConnections(domain)[0];
+        address storedReplicaImpl = vm
+            .load(address(replicaOfUpgrade(domain, remote).beacon), bytes32(0))
+            .bytes32ToAddress();
+        assertEq(storedReplicaImpl, address(replicaImpl));
+
+        // Verify Gov Router implementation
+        address storedGovRouterImpl = vm
+            .load(address(governanceRouterUpgrade(domain).beacon), bytes32(0))
+            .bytes32ToAddress();
+        assertEq(storedGovRouterImpl, address(govRouterImpl));
+
+        // Verify Bridge Router implementation
+        address storedBridgeRouterImpl = vm
+            .load(address(bridgeRouterUpgrade(domain).beacon), bytes32(0))
+            .bytes32ToAddress();
+        assertEq(storedBridgeRouterImpl, address(bridgeRouterImpl));
+
+        // Verify Bridge Token Implementation
+        address storedBridgeTokenImpl = vm
+            .load(address(bridgeTokenUpgrade(domain).beacon), bytes32(0))
+            .bytes32ToAddress();
+        assertEq(storedBridgeTokenImpl, address(bridgeTokenImpl));
+
+        // Verify Bridge Token Implementation
+        address storedTokenRegistryImpl = vm
+            .load(address(tokenRegistryUpgrade(domain).beacon), bytes32(0))
+            .bytes32ToAddress();
+        assertEq(storedTokenRegistryImpl, address(tokenRegistryImpl));
+    }
+
+    /// @notice Read the governance actions that we created during the Reboot
+    /// @dev The gov actions are expected to be stored in a JSON file of the form
+    /// "<domain_name>.json" in "actions/"
+    function readGovActions() internal returns (Batch memory) {
+        string memory fileName = string(
+            abi.encodePacked("actions/", domain, ".json")
+        );
+        string memory file = vm.readFile(fileName);
+        string memory builtKey = string(
+            abi.encodePacked(
+                ".built.",
+                vm.toString(uint256(domainNumber)),
+                ".data"
+            )
+        );
+        builtActions = abi.decode(file.parseRaw(builtKey), (bytes));
+        // Not required for now, but useful to have
+        // localActions = abi.decode(file.parseRaw(".local"), (Batch[]));
+        // remoteActions = abi.decode(file.parseRaw(".remote"), (Batch[]));
+    }
+
+    /// @notice Execute the built governance actions in recovery mode
+    function executeGovActions() internal {
+        GovernanceRouter govRouter = getGovernanceRouter(domain);
+        vm.prank(govRouter.recoveryManager());
```

Comment:
could do `if(inRecovery) startPrank(recoveryManager) else startPrank(governor)` but not necessary atm